packages/components/src/utils/hds-carbon-icon-map.ts

```diff
@@ -0,0 +1,3453 @@
+import { HdsIconSizeValues } from '../components/hds/icon/types.ts';
+
+import type { IconName } from '@hashicorp/flight-icons/svg';
+import type { HdsIconSizes } from '../components/hds/icon/types.ts';
+
+export interface CarbonIconAttrs {
+  viewBox: string;
+  width: number | string;
+  height: number | string;
+}
+
+export type CarbonIconNodeAttrs = {
+  width: number;
+  height: number;
+  d?: string;
+  cx?: number | string;
+  cy?: number | string;
+  r?: number | string;
+  x?: number | string;
+  y?: number | string;
+  points?: string;
+  opacity?: number | string;
+  transform?: string;
+  fill?: string;
+  stroke?: string;
+  'stroke-width'?: number | string;
+  'stroke-linecap'?: string;
+  'stroke-linejoin'?: string;
+  'stroke-miterlimit'?: string;
+  // allow anything else but still narrow to AttrValue-like types
+  [key: string]: string | number | undefined;
+};
+
+export interface CarbonIconContent {
+  elem: string;
+  attrs: CarbonIconNodeAttrs;
+}
+
+export interface CarbonIcon {
+  attrs: CarbonIconAttrs;
+  content: CarbonIconContent[];
+  name: string;
+  size: string | number;
+}
+
+export interface IconMapItem {
+  hdsIconName: IconName;
+  carbonIconName: string;
+  importCarbonIcon: (size?: HdsIconSizes) => Promise<unknown>;
+}
+
+export function getByHdsIconName(
+  hdsIconName: IconName
+): IconMapItem | undefined {
+  return hdsCarbonIconList.find((item) => item.hdsIconName === hdsIconName);
+}
+
+export const hdsCarbonIconList: IconMapItem[] = [
+  {
+    hdsIconName: 'accessibility',
+    carbonIconName: 'accessibility--alt',
+    importCarbonIcon: async (
+      size: HdsIconSizes = HdsIconSizeValues.Sixteen
+    ) => {
+      return size === HdsIconSizeValues.Sixteen
+        ? await import('@carbon/icons/es/accessibility--alt/16')
+        : await import('@carbon/icons/es/accessibility--alt/24');
```

Comment:
[question] would something like this work? 
```suggestion
return await import(`@carbon/icons/es/accessibility--alt/${size}`)
```

Usually this doesn't work for tree shakings, but in this case (as we discussed) is not really a tree-shaking done at build time, more of a dynamic import at run time of tiny bundled code?